app/src/main/java/com/ichi2/apisample/helper/AnkiDroidHelper.java

```diff
@@ -323,21 +331,91 @@ public Long addNewDeck(String deckName) {
 
     // @todo: refactor once new version release of "com.ichi2.anki.api" is available
     public String addFileToAnkiMedia(String uriString) {
+        Uri uri = Uri.parse(uriString);
+        mContext.grantUriPermission("com.ichi2.anki", uri, Intent.FLAG_GRANT_READ_URI_PERMISSION);
+        String type = mResolver.getType(uri);
+        File tempAudioFile = null;
+        if (type.startsWith("video")) {
+            if (!type.equals("video/mp4")) {
+                mContext.revokeUriPermission(uri, Intent.FLAG_GRANT_READ_URI_PERMISSION);
+                return null;
+            }
+            try {
+                MediaMuxer mediaMuxer = null;
+                MediaExtractor mediaExtractor = new MediaExtractor();
+                final String tempAudioPath = Environment.getExternalStorageDirectory().getPath() + "/tempOutput.mp3";
+                mediaExtractor.setDataSource(mContext, uri, null);
+                int trackCount = mediaExtractor.getTrackCount();
+                int audioTrackindex = -1;
+                int framerate = -1;
+                for (int i = 0; i < trackCount; i++) {
+                    MediaFormat trackFormat = mediaExtractor.getTrackFormat(i);
+                    String mime = trackFormat.getString(MediaFormat.KEY_MIME);
+                    if (mime.startsWith("audio/")) {
+                        mediaExtractor.selectTrack(i);
+                        mediaMuxer = new MediaMuxer(tempAudioPath, MediaMuxer.OutputFormat.MUXER_OUTPUT_MPEG_4);
+                        audioTrackindex = mediaMuxer.addTrack(trackFormat);
+                        mediaMuxer.start();
+                    }
+                    if (mime.startsWith("video/")) {
+                        framerate = trackFormat.getInteger(MediaFormat.KEY_FRAME_RATE);
+                        break;
+                    }
+                }
+
+                if (mediaMuxer == null) {
+                    mContext.revokeUriPermission(uri, Intent.FLAG_GRANT_READ_URI_PERMISSION);
+                    return null;
+                }
+
+                MediaCodec.BufferInfo bufferInfo = new MediaCodec.BufferInfo();
+                bufferInfo.presentationTimeUs = 0;
+
+                ByteBuffer byteBuffer = ByteBuffer.allocate(500 * 1024);
```

Comment:
Please use some contants for "magic numbers" or describe in comments what is that

Reply:
Fixed